render/tests/test-component.js

```diff
@@ -12,586 +12,1270 @@ o.spec("component", function() {
-						{tag: "label"},
-						{tag: "input"},
-					]
-				}
-			}
-			render(root, [{tag: component}])
-			
-			o(root.childNodes.length).equals(2)
-			o(root.childNodes[0].nodeName).equals("LABEL")
-			o(root.childNodes[1].nodeName).equals("INPUT")
-		})
-		o("can return string", function() {
-			var component = {
-				view: function(vnode) {
-					return "a"
-				}
-			}
-			render(root, [{tag: component}])
-			
-			o(root.firstChild.nodeType).equals(3)
-			o(root.firstChild.nodeValue).equals("a")
-		})
-		o("can return falsy string", function() {
-			var component = {
-				view: function(vnode) {
-					return ""
-				}
-			}
-			render(root, [{tag: component}])
-			
-			o(root.firstChild.nodeType).equals(3)
-			o(root.firstChild.nodeValue).equals("")
-		})
-		o("can return number", function() {
-			var component = {
-				view: function(vnode) {
-					return 1
-				}
-			}
-			render(root, [{tag: component}])
-			
-			o(root.firstChild.nodeType).equals(3)
-			o(root.firstChild.nodeValue).equals("1")
-		})
-		o("can return falsy number", function() {
-			var component = {
-				view: function(vnode) {
-					return 0
-				}
-			}
-			render(root, [{tag: component}])
-			
-			o(root.firstChild.nodeType).equals(3)
-			o(root.firstChild.nodeValue).equals("0")
-		})
-		o("can return boolean", function() {
-			var component = {
-				view: function(vnode) {
-					return true
-				}
-			}
-			render(root, [{tag: component}])
-			
-			o(root.firstChild.nodeType).equals(3)
-			o(root.firstChild.nodeValue).equals("true")
-		})
-		o("can return falsy boolean", function() {
-			var component = {
-				view: function(vnode) {
-					return false
-				}
-			}
-			render(root, [{tag: component}])
-			
-			o(root.firstChild.nodeType).equals(3)
-			o(root.firstChild.nodeValue).equals("false")
-		})
-		o("can return null", function() {
-			var component = {
-				view: function(vnode) {
-					return null
-				}
-			}
-			render(root, [{tag: component}])
-			
-			o(root.childNodes.length).equals(0)
-		})
-		o("can return undefined", function() {
-			var component = {
-				view: function(vnode) {
-					return undefined
-				}
-			}
-			render(root, [{tag: component}])
-			
-			o(root.childNodes.length).equals(0)
-		})
-		o("can update when returning fragments", function() {
-			var component = {
-				view: function(vnode) {
-					return [
-						{tag: "label"},
-						{tag: "input"},
-					]
-				}
-			}
-			render(root, [{tag: component}])
-			render(root, [{tag: component}])
-			
-			o(root.childNodes.length).equals(2)
-			o(root.childNodes[0].nodeName).equals("LABEL")
-			o(root.childNodes[1].nodeName).equals("INPUT")
-		})
-		o("can update when returning primitive", function() {
-			var component = {
-				view: function(vnode) {
-					return "a"
-				}
-			}
-			render(root, [{tag: component}])
-			render(root, [{tag: component}])
-			
-			o(root.firstChild.nodeType).equals(3)
-			o(root.firstChild.nodeValue).equals("a")
+	o.spec("object", function() {
```

Comment:
This section is just the existing tests, and the only change is indentation. You can feel free to skip this part.